### ISSUE TITLE
Change behavior for empty values when deserializing an option

### ISSUE THREAD

**Issue by jplatte** [jplatte/serde_html_form#13]
Currently, the following two forms get the `foo` field deserialized to `None` given the input `foo=`:

```rust
#[derive(Deserialize)]
struct FormA {
    foo: Option<String>,
}

#[derive(Deserialize)]
struct FormB {
   foo: Option<Vec<String>>,
}
```

I think the most reasonable output would instead be an empty string (as the only list element) wrapped in `Some` for both forms.

What might also be relevant: I think the same forms with `String` replaced by `f64` should produce `foo: None` for `FormA` and a deserialization error for `FormB`. I think both should already be the case.

This is a breaking change, so if implemented I will publish it as part of `serde_html_form 0.3.0`.

**Comment by jplatte** [comment]
Should also when working on this consider testing `?foo` separately from `?foo=`. Both should produce an empty string (see #21).

**Comment by Turbo87** [comment]
@jplatte do you have any clue how to implement this special casing of `String` vs. the other types? I don't have a lot of experience with serde internals, but all my experiments so far have failed... 😅

**Comment by jplatte** [comment]
No, I don't know either. Writing serde deserializers is *hard*.

**Comment by Turbo87** [comment]
@dtolnay you don't happen to have any suggestions? :D

**Comment by KrullBorg** [comment]
I think i have the same problem: i need to distinguish if a field is present into the form from if it is empty. Is there some workaround?

**Comment by jplatte** [comment]
`#[serde]` derive attributes are not something a (de)serializer has to add special support for. If you're having problems with that, please open a separate issue.

**Comment by jplatte** [comment]
So, the current behavior has some pretty convincing background [here](https://github.com/jplatte/serde_html_form/pull/4#issuecomment-1359024420), and was introduced [here](https://github.com/jplatte/serde_html_form/pull/5) (cc @thomaseizinger).

I've attempted to make serde do what I want, but am now pretty confident that serde's data model just doesn't allow for this distinction to be made. _However_, I think we can get close enough by keying off of the value type produced by the `Visitor` impl used in `deserialize_option` - something that will require some dirty tricks but should be sound. I have opened https://github.com/sagebind/castaway/issues/38 about this.
